### PR TITLE
Publish Action Fix

### PR DESCRIPTION
**Pull request by ericfichtel**
## Description

There is currently an error in while attempting to publish the newest release to npm using the `npm-publish` action: 
https://github.com/adobe/target-nodejs-sdk/actions/runs/3886422538/jobs/6633871334

Proposing the removal of the `token` while evoking `actions/checkout@master` and switching to `actions/checkout@v2` to align with the other SDKs. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: .github/workflows/npm-publish.yml [hunk 1]
```diff
@@ -12,23 +12,18 @@ jobs:
     name: npm-publish
     runs-on: ubuntu-latest
     steps:
-      - name: Checkout repository
```
Comment (jasonwaters) [inline .github/workflows/npm-publish.yml:15]: is there a reason you removed the name?

File: .github/workflows/npm-publish.yml [hunk 2]
```diff
@@ -12,23 +12,18 @@ jobs:
...
-        uses: actions/checkout@master
+      - uses: actions/checkout@v2
         with:
           fetch-depth: 0
-          token: ${{ secrets.ADOBE_BOT_GITHUB_TOKEN }}
```
Comment (jasonwaters) [inline .github/workflows/npm-publish.yml:19]: token is no longer needed?

Comment (ericfichtel) [reply .github/workflows/npm-publish.yml:19]: I believe this line is what is causing the release action to fail. It does not appear that any of the actions across the SDKs use a token while calling the checkout action.

File: .github/workflows/npm-publish.yml (hunk 1 above)
Comment (ericfichtel) [reply .github/workflows/npm-publish.yml:15]: I removed the name to keep it consistent with `build.yml` and what the format is for the Java SDK actions. It is not necessary to remove it though.

File: .github/workflows/npm-publish.yml (hunk 2 above)
Comment (ericfichtel) [reply .github/workflows/npm-publish.yml:19]: Unless there is a different security configuration for this repo than the others. However, the `CI` action of this repo does not use a token during checkout.

**Review by dcottingham — APPROVED** [review]
Let's give it a try

**Pull request merged**